### PR TITLE
Enable ethereum pending txs

### PR DESCRIPTION
**Pull request by hewigovens**
Changes:
1. Remove pending tx condition
2. Check data field when fill metadata
3. Calculate gas (limit * price) for pending tx
4. Add pending incoming eth/token transfer test

### CODE REVIEW

File: platform/ethereum/blockbook/transaction_test.go [hunk 1]
```diff
@@ -182,6 +217,23 @@ func TestNormalizePage(t *testing.T) {
...
+					"id": "0xfea3444def47c77c5ede0f705d1bc84bb258d51fa454971985c181d6bc679144",
+					"coin": 60,
+					"from": "0x7d8bf18C7cE84b3E175b339c4Ca93aEd1dD166F1",
+					"to": "0x47331175b23C2f067204B506CA1501c26731C990",
+					"fee": "0",
+					"date": 1589249442,
+					"block": 0,
+					"status": "pending",
+					"sequence": 0,
+					"type": "contract_call",
```
Comment (vikmeup) [inline platform/ethereum/blockbook/transaction_test.go:230]: why does it says contract call?

File: platform/ethereum/blockbook/transaction_test.go [hunk 2]
```diff
@@ -182,6 +217,23 @@ func TestNormalizePage(t *testing.T) {
 						"from": "0x7d8bf18C7cE84b3E175b339c4Ca93aEd1dD166F1",
 						"to": "0x7d8bf18C7cE84b3E175b339c4Ca93aEd1dD166F1"
 					}
+				  },{
+					"id": "0xfea3444def47c77c5ede0f705d1bc84bb258d51fa454971985c181d6bc679144",
+					"coin": 60,
+					"from": "0x7d8bf18C7cE84b3E175b339c4Ca93aEd1dD166F1",
+					"to": "0x47331175b23C2f067204B506CA1501c26731C990",
+					"fee": "0",
```
Comment (vikmeup) [inline platform/ethereum/blockbook/transaction_test.go:225]: can we include fee here? as gas price * gas limit

**Comment by hewigovens** [comment]
@vikmeup current blockbook api returns pending tx like this
```json
{
    "txid": "0xf1fc796cab0a025775658c41b7a77556528081a4141a3dcc821e4c60d8fcf9cc",
    "vin": [
        {
            "n": 0,
            "addresses": [
                "0x23a80893B3CE3B17598c2077739559B5f13F81bA"
            ],
            "isAddress": true
        }
    ],
    "vout": [
        {
            "value": "0",
            "n": 0,
            "addresses": [
                "0xdAC17F958D2ee523a2206206994597C13D831ec7"
            ],
            "isAddress": true
        }
    ],
    "blockHeight": -1,
    "confirmations": 0,
    "blockTime": 1589251274,
    "value": "0",
    "fees": "0",
    "rbf": true,
    "tokenTransfers": [
        {
            "type": "ERC20",
            "from": "0x23a80893B3CE3B17598c2077739559B5f13F81bA",
            "to": "0x87c574FB2DF0EaA2dAf5fc4a8A16dd3Ce39011B1",
            "token": "0xdAC17F958D2ee523a2206206994597C13D831ec7",
            "name": "Tether USD",
            "symbol": "USDT",
            "decimals": 6,
            "value": "3940610000"
        }
    ],
    "ethereumSpecific": {
        "status": -1,
        "nonce": 0,
        "gasLimit": 100000,
        "gasUsed": null,
        "gasPrice": "30000000000"
    }
}
```

**Comment by hewigovens** [comment]
we couldn't get nonce/data/gas, so the type is fallback to contract_call

**Comment by vikmeup** [comment]
Can we adjust logic to support this? since we have all the data?

File: platform/ethereum/blockbook/transaction_test.go (hunk 2 above)
Comment (hewigovens) [reply platform/ethereum/blockbook/transaction_test.go:225]: this one can be fixed

**Comment by hewigovens** [comment]
need to include abi parsing somehow, not a quick fix, also we need to update to latest code to include data field

File: platform/ethereum/blockbook/transaction_test.go (hunk 2 above)
Comment (hewigovens) [reply platform/ethereum/blockbook/transaction_test.go]: done

File: platform/ethereum/blockbook/transaction_test.go (hunk 1 above)
Comment (hewigovens) [reply platform/ethereum/blockbook/transaction_test.go:316]: this test case is correct now, data: `0xdeadbeef`

**Comment by hewigovens** [comment]
> Can we adjust logic to support this? since we have all the data?

it turns out we don't need to do abi parsing now, the only issue left is low gas price tx might be dropped (but can be seen in etherscan.io)

File: platform/ethereum/blockbook/model.go [hunk 3]
```diff
@@ -73,4 +73,5 @@ type EthereumSpecific struct {
 	GasLimit *big.Int `json:"gasLimit"`
 	GasUsed  *big.Int `json:"gasUsed"`
 	GasPrice string   `json:"gasPrice"`
+	Data     string   `json:"data,omitempty"`
```
Comment (EnoRage) [inline platform/ethereum/blockbook/model.go:76]: It could be empty?

File: platform/ethereum/blockbook/model_extension.go [hunk 4]
```diff
@@ -24,6 +26,21 @@ func (t *Transaction) FromAddress() string {
 	return ""
 }
 
+func (t *Transaction) GetFee() string {
+	status, _ := t.EthereumSpecific.GetStatus()
```
Comment (EnoRage) [inline platform/ethereum/blockbook/model_extension.go:30]: can you handle error here please

File: platform/ethereum/blockbook/model.go (hunk 3 above)
Comment (hewigovens) [reply platform/ethereum/blockbook/model.go:76]: Yes, this field is added recently, we may switch to old node which doesn’t return data

File: platform/ethereum/blockbook/model_extension.go (hunk 4 above)
Comment (hewigovens) [reply platform/ethereum/blockbook/model_extension.go:30]: Looks weird, but _ is not an error but status string, can be improved tho

File: platform/ethereum/blockbook/model_extension.go [hunk 5]
```diff
@@ -24,6 +26,21 @@ func (t *Transaction) FromAddress() string {
...
+	if status != blockatlas.StatusPending {
+		return t.Fees
+	}
```
Comment (vikmeup) [inline platform/ethereum/blockbook/model_extension.go:33]: can you add test for this specific function?

Comment (hewigovens) [reply platform/ethereum/blockbook/model_extension.go:33]: I think it's already covered in `TestNormalizePage`, we have both pending and completed txs

**Pull request merged**